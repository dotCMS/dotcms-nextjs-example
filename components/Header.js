import React from 'react';


import Nav from './layout/Nav/Nav'

function Header() {

    return (
        <Nav />
    );
}

export default Header;
