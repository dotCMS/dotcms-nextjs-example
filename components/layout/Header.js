import Nav from './Nav/Nav';
import LocaleDropdown from './LocaleDropdown';
import RouterLink from '../RouterLink';

const Header = () => {
    return (
        <>
            <header>
                {/* RD Navbar */}
                <div className="rd-navbar-wrap">
                    <nav
                        className="rd-navbar rd-navbar-static rd-navbar-classic"
                        data-layout="rd-navbar-fixed"
                        data-sm-layout="rd-navbar-fixed"
                        data-md-layout="rd-navbar-fixed"
                        data-md-device-layout="rd-navbar-fixed"
                        data-lg-layout="rd-navbar-static"
                        data-lg-device-layout="rd-navbar-static"
                        data-xl-layout="rd-navbar-static"
                        data-xl-device-layout="rd-navbar-static"
                        data-lg-stick-up-offset="78px"
                        data-xl-stick-up-offset="98px"
                        data-xxl-stick-up-offset="118px"
                        data-lg-stick-up="true"
                        data-xl-stick-up="true"
                        data-xxl-stick-up="true"
                    >
                        <div
                            className="rd-navbar-collapse-toggle rd-navbar-fixed-element-1"
                            data-rd-navbar-toggle=".rd-navbar-collapse"
                        >
                            <span></span>
                        </div>

                        <div className="rd-navbar-aside-outer rd-navbar-collapse bg-default">
                            <div className="rd-navbar-aside">
                                {/* CONTACT INFO */}
                                <ul className="list-inline list-inline-xxl">
                                    <li>
                                        <div className="unit unit-spacing-xs">
                                            <div className="unit-left">
                                                <span className="icon mdi mdi-map-marker"></span>
                                            </div>
                                            <div className="unit-body">
                                                <a href="#">
                                                    3059 Grand Avenue, Suite 300 Miami, FL, 33133
                                                    USA
                                                </a>
                                            </div>
                                        </div>
                                    </li>
                                    <li>
                                        <div className="unit unit-spacing-xs">
                                            <div className="unit-left">
                                                <span className="icon mdi mdi-cellphone-android"></span>
                                            </div>
                                            <div className="unit-body">
                                                <a href="tel:+1-888-404-6185">+1-888-404-6185</a>
                                            </div>
                                        </div>
                                    </li>
                                    <li>
                                        <div className="unit unit-spacing-xs">
                                            <div className="unit-left">
                                                <span className="icon mdi mdi-email-outline"></span>
                                            </div>
                                            <div className="unit-body">
                                                <a href="mailto: info@dotcms.com">
                                                    info@dotcms.com
                                                </a>
                                            </div>
                                        </div>
                                    </li>
                                </ul>
                                <div>
                                    <LocaleDropdown />

                                    {/* CART */}
                                    {/* <a className="basket" href="/store/cart">
                                        <span className="icon mdi mdi-cart-outline"></span>
                                        <span className="basket-count">3</span>
                                    </a> */}
                                </div>
                            </div>
                        </div>

                        <div className="rd-navbar-main-outer">
                            <div className="rd-navbar-main">
                                <div className="rd-navbar-panel">
                                    <button
                                        className="rd-navbar-toggle"
                                        data-rd-navbar-toggle=".rd-navbar-nav-wrap"
                                        aria-label="Toggle Menu"
                                    >
                                        <span></span>
                                    </button>
                                    <div className="rd-navbar-brand">
                                        <RouterLink href="/index" className="brand">
                                            <img
                                                className="brand-logo-dark"
                                                src={`/application/themes/travel/images/logo.png`}
                                                width="195"
                                                height="40"
                                                alt="TravelLux Home"
                                            />
                                        </RouterLink>
                                    </div>
                                </div>

                                {/* NAV BAR */}
                                <Nav />
                            </div>
                        </div>
                    </nav>
                </div>
            </header>
        </>
    );
};

export default Header;
