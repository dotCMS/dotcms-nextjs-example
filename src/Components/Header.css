header.header__menu {
    background: transparent !important;
    justify-content: space-between;
    z-index: 1;
}

.header__extend {
    align-items: center;
    background-image: url(../theme/images/cover_img_1.jpg);
    background-size: cover;
    background-position: center center;
    background-attachment: fixed;
    background-repeat: no-repeat;
    display: flex;
    height: var(--header-height);
    justify-content: center;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: 0;
}

.header__overlay {
    background: linear-gradient(135deg, #499bea 0%, #798eea 100%);
    bottom: 0;
    height: var(--header-height);
    left: 0;
    opacity: 0.95;
    position: absolute;
    right: 0;
    top: 0;
}

.header__brand-logo.navbar-brand {
    color: var(--white);
    font-size: 1.25rem;
    font-weight: 700;
    text-transform: uppercase;
}

.header__page-title,
.header__breadcrumbs {
    color: var(--white);
    position: relative;
}

.header__page-title {
    font-size: 3.75rem;
    font-weight: 300;
    margin-bottom: 1.25rem;
    text-align: center;
}

@media screen and (max-width: 768px) {
    .header__page-title {
        font-size: 2.125rem;
    }
}
