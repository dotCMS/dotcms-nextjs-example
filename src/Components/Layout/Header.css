header.header__menu {
    display: block;
    z-index: 1;
}

header.header__menu > button {
    float: right;
}

.header{
    background-color:black;
    margin-bottom:0px;
    padding-bottom:0px;
    z-index: 999;
}

.navbar-nav{
    justify-content:flex-end;
}


.navbar-brand {
    text-align: left;
    color: var(--white);
    font-size: 1.25rem;
    font-weight: 700;
    text-transform: uppercase;
}


.header__breadcrumbs {
    color: var(--white);
    position: relative;
}

.layout__content {
    margin-top:0px!important;


}
