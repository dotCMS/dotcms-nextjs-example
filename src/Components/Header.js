import React from 'react';
import { Navbar, NavbarBrand } from 'reactstrap';
import Nav from './Nav/Nav';
import './Header.css';

const Header = (props) => {
    return (
        <>
            <Navbar tag="header" className="header__menu container" expand="md">
                <NavbarBrand className="header__brand-logo" href="/">
                    DotCMS - SPA
                </NavbarBrand>
                <Nav />
            </Navbar>
            <section className="header__extend">
                <div className="header__overlay" />
                <div className="header__content">
                    <h2 className="header__page-title">{props.title}</h2>
                </div>
            </section>
        </>
    );
};

export default Header;
