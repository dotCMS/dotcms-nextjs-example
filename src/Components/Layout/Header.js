import React from 'react';
import { Navbar, NavbarBrand, Container as BootstrapContainer } from 'reactstrap';
import { Link } from 'react-router-dom';

import Nav from '../Nav/Nav';
import './Header.css';


const Header = (props) => {
    return (
        <>
            <div className="header">
                <Navbar tag="header" className="container" expand="md" >
                    <NavbarBrand className="header__brand-logo" tag={Link} to="/">
                        dotCMS SPA
                        <div style={{fontSize:'14px',marginLeft:'7px'}}>{props.title}</div>
                    </NavbarBrand>
                    
                    <Nav />
                </Navbar>
            </div>
        </>
    );
};

export default Header;
