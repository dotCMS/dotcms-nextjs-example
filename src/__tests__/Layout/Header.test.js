import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Header from '../../Components/Layout/Header';

describe('<Header />', () => {
    it('renders and matches snapshot', () => {
        const wrapper = shallow(<Header />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
