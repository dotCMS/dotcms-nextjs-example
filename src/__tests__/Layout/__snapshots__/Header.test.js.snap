// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders and matches snapshot 1`] = `
<Fragment>
  <Navbar
    className="header__menu container"
    expand="md"
    tag="header"
  >
    <NavbarBrand
      className="header__brand-logo"
      tag={[Function]}
      to="/"
    >
      DotCMS - SPA
    </NavbarBrand>
    <Nav />
  </Navbar>
  <section
    className="header__extend"
  >
    <div
      className="header__overlay"
    />
    <div
      className="header__content"
    >
      <Container
        tag="div"
      >
        <h2
          className="header__page-title"
        />
      </Container>
    </div>
  </section>
</Fragment>
`;
